src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,473 @@ const ActorInit En_Heishi1_InitVars = {
+    s16 frameCount = (f32)SkelAnime_GetFrameCount(&D_06005880.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005880, 3.0f, 0.0f, frameCount, 0, -3.0f);
+    this->bodyTurnSpeed = 0.0f;
+    this->moveSpeed = 0.0f;
+    func_8010B680(globalCtx, 0x702D, &this->actor);
+    Interface_SetDoAction(globalCtx, 0x12);
+    this->actionFunc = EnHeishi1_MoveToLink;
+}
+
+void EnHeishi1_MoveToLink(EnHeishi1* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothScaleMaxF(&this->actor.posRot.pos.x, player->actor.posRot.pos.x, 1.0f, this->moveSpeed);
+    Math_SmoothScaleMaxF(&this->actor.posRot.pos.z, player->actor.posRot.pos.z, 1.0f, this->moveSpeed);
+    Math_SmoothScaleMaxF(&this->moveSpeed, 6.0f, 1.0f, 0.4f);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 3, this->bodyTurnSpeed, 0);
+    Math_SmoothScaleMaxF(&this->bodyTurnSpeed, 3000.0f, 1.0f, 300.0f);
+    Math_SmoothDownscaleMaxF(&this->headAngle, 0.5f, 2000.0f);
+
+    if (this->actor.xzDistFromLink < 70.0f) {
+        this->actionFunc = EnHeishi1_SetupTurnTowardLink;
+    }
+}
+
+void EnHeishi1_SetupWait(EnHeishi1* this, GlobalContext* globalCtx) {
+    s16 rand;
+    s16 frameCount = (f32)SkelAnime_GetFrameCount(&D_06005C30.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, this->animSpeed, 0.0f, frameCount, 0, this->transitionRate);
+    this->headBehaviorDecided = false;
+    this->headDirection = Math_Rand_ZeroFloat(1.99f);
+    rand = Math_Rand_ZeroFloat(50.0f);
+    this->waitTimer = rand + 50;
+    this->actionFunc = EnHeishi1_Wait;
+}
+
+void EnHeishi1_Wait(EnHeishi1* this, GlobalContext* globalCtx) {
+    s16 randOffset;
+    s32 i;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (!sPlayerIsCaught) {
+        switch (this->headBehaviorDecided) {
+            case false:
+                this->headDirection++;
+                // if headDirection is odd, face 52 degrees left
+                this->headAngleTarget = (this->headDirection & 1) ? 9472.0f : -9472.0f;
+                randOffset = Math_Rand_ZeroFloat(30.0f);
+                this->headTimer = sBaseHeadTimers[this->type] + randOffset;
+                this->headBehaviorDecided = true;
+                break;
+            case true:
+                if (this->headTimer == 0) {
+                    if (this->waitTimer == 0) {
+                        if ((this->type == 0) || (this->type == 1)) {
+                            this->waypoint++;
+                            if (this->waypoint >= 4) {
+                                this->waypoint = 0;
+                            }
+                        } else {
+                            // waypoints are defined with corners as 0-3 and middle points as 4-7
+                            // to choose the next waypoint they hardcoded the order "04152637" in an array
```

Comment:
I don't think we should be referring to og devs in 3rd person like that in documentation comments. It's better to explain what the code does in an impersonal way imo.
```suggestion
                            // to choose the next waypoint, the order "04152637" is hardcoded in an array
```